README reworked (#363)
TheAssassin/AppImageLauncher: README.md

# AppImageLauncher
makes your Linux desktop AppImage ready™
<p align="center">
<img src="https://github.com/TheAssassin/AppImageLauncher/raw/master/resources/icons/hicolor/128x128/apps/AppImageLauncher.png"/>
</p>

> Integrate AppImages to your application launcher with one click, wherefrom you can manage, update and remove them.

Author: ```suggestion
Integrate AppImages to your application launcher with one click, and manage, update and remove them from there.
```

<img src="https://github.com/TheAssassin/AppImageLauncher/raw/master/resources/icons/hicolor/128x128/apps/AppImageLauncher.png"/>
</p>

> Integrate AppImages to your application launcher with one click, wherefrom you can manage, update and remove them.

Reviewer: Wherefrom is really archaic, and doesn't work well in a tech context.

Author: How is it any different from "furthermore"?

Reviewer: Furthermore is not an archaic term as far as I can see, it's modern English. And it's not a synonym for "wherefrom".

Will combine your suggestions with my own change requests, create some sort of result and push it tonight.

Author: No don't do that, let's keep it within reason of a snimple snuppler like myself.

<img src="https://github.com/TheAssassin/AppImageLauncher/raw/master/resources/icons/hicolor/128x128/apps/AppImageLauncher.png"/>
</p>

> Installation makes your Linux desktop AppImage ready™, and you can double-click AppImages again without making them executable first.

Author: I don't like the advertisement angle. It is still there, and it doesn't do anything.

> Installation makes your Linux desktop AppImage ready™, and you can double-click AppImages again without making them executable first.

Reviewer: This sentence was quoted in many articles. It may be advertising, but what's wrong with that?

> Installation makes your Linux desktop AppImage ready™, and you can double-click AppImages again without making them executable first.

Author: Reduced rank for any new articles. The main problem is it doesn't carry anything.
Use it somewhere and in some manner that signifies it is a slogan.
Go all the way with a nice SVG or something.

> Installation makes your Linux desktop AppImage ready™, and you can double-click AppImages again without making them executable first.

Reviewer: > Go all the way with a nice SVG or something. Presumably you don't hold a trademark for it?

I don't have to? It's also not meant too seriously.

I don't understand this "either you don't use it at all or you must design some sort of SVG". It makes no sense to me. Let's keep it as is.

> Installation makes your Linux desktop AppImage ready™, and you can double-click AppImages again without making them executable first.

Author: The point is to not be cheap with it. To make that concern count, everything has to be in order, and it has to be done right. Otherwise it ages quick.

> Installation makes your Linux desktop AppImage ready™, and you can double-click AppImages again without making them executable first.

Reviewer: I don't understand what you mean.

You can integrate AppImages with a single mouse-click, and manage them from your application launcher.

> Updating and removing AppImages can also be done from there.

Author: Changed the whole thing around.

**Traditional packages are highly recommended *if possible*, as they provide many more features and provide a much better overall experience.**

## About the project

AppImages and Linux desktops don't currently work very well together. Since AppImages are normal executables, it'd suffice if desktop environments like Plasma, GNOME, Xfce, ... would assist users in making those files executable, but as we learned recently, some desktop environments consider this a security risk, and would rather users use app stores.

The ability to run isn't all that's needed to provide a good desktop experience. AppImages should be accessible from the application menus and launchers. This so-called "desktop integration" can't be provided by the AppImages themselves, even though some AppImages ship with a "desktop integration script" prompting the user to do so. There are too many impliciations requiring external software, especially regarding cleanup and removal of AppImages. If applications are simply made executable, they're still spread all over the users' personal files and folders.

> The average user doesn't necessarily like a Downloads directory full of AppImages with cryptic filenames.

Author: I like it, because it points out clearly what otherwise would be the case, and makes a point out of how this is a thing for regular users.

Therefore, new, system-side solutions have been developed to perform the desktop integration.
The oldest available solution is [appimaged](https://github.com/AppImage/AppImageKit), a daemon users can install that performs everything in the background, automagically, without notifying the user in any way.

> It scans a predefined set of directories including `~/Downloads` and `~/.bin`, makes AppImages which are found executable and performs the desktop integration. This is rather inefficient, as appimaged's operations and monitoring produce a lot of file I/O. Also, many users don't like the lack of control. (It has been deprecated in favour of go-appimage.)

Author: > It scans a predefined set of directories including `~/Downloads` and `~/.bin`, makes AppImages which are found executable

?

> AppImageLauncher is a new solution that integrates deeply in the system and intercepts all attempts to open an AppImage, becoming the first instance to handle all AppImage invocations.

Author: This section could be cut, as it is arguable replicated above. It is OK for people writing articles.

Reviewer: I'd rather keep it this paragraph, but would split the sentence in half:

```suggestion
AppImageLauncher is a new solution that integrates deeply in the system.
It intercepts all attempts to open an AppImage, [...]
```

The second sentence should end in something like "takes responsiblity for the AppImage process", "handles them during their lifetime" etc. That'd also allow to dump the following sentence.

Author: "integrates deeply in the system" sounds quite bad, and it would be "into". Going "with".

Reviewer: To invoke, to execute, to run, they're all pretty much synonyms in this context.

Being the launcher for AppImages, AppImageLauncher can control how the system treats AppImages.

> It can perform the desktop integration, AppImage removal (also called "uninstallation" sometimes, but as AppImages are not really installed, this term doesn't fit very well), and could be used for many other tasks in the future, like update checks and the like.

Reviewer: Updating is actually available now. The sentence needs an update.

A few articles have been written about AppImageLauncher already:

> English

Reviewer: I would like to keep them in the order how I discovered them. Also, the title "English" makes little sense, as the other languages are not mentioned. Let's keep the suffixes.

  - https://www.linuxuprising.com/2018/04/easily-run-and-integrate-appimage-files.html
  - https://www.freeyourdesktop.com/2018/07/install-manage-appimages-with-appimagelauncher/
    - same article also available here: https://medium.com/@freeyourdesktopblog/install-manage-appimages-with-appimagelauncher-2a2078c55f37
  - https://www.freeyourdesktop.com/2018/07/install-manage-appimages-with-appimagelauncher/ (English)

>   - please note that AppImageLauncher could *not* be "installed" via AppImage at that time, only recently we added a Lite version that now can be installed from an AppImage (more information will follow soon!)

Reviewer: It's a comment for this specific article. Removing the indentation is not a good idea.

  - https://www.reallinuxuser.com/15-best-things-to-do-after-installing-zorin-os-15/ (English)
  - https://manjaro.site/how-to-install-appimage-launcher-on-ubuntu-19-04/ (English)
  - https://cubiclenate.com/2020/01/09/appimagelauncher-appimage-manager-on-opensuse/ (English)
    - see also: https://cubiclenate.com/2020/01/10/noodlings-lighting-the-emby-server-with-kdenlive/
  - https://chrisrmiller.com/2019/05/29/integrate-appimages-into-your-linux-distro/ (English)
  - https://medium.com/@freeyourdesktopblog/install-manage-appimages-with-appimagelauncher-2a2078c55f37 (English)

>   

Reviewer: See above. This makes no sense.

  - https://dolys.fr/forums/topic/gerer-les-appimage-sous-linux/ (French)
  - https://linux-user.gr/t/eykolh-chrhsh-efarmogwn-appimage/2066 (Greek)
  - https://linuxmint.hu/blog/2018/12/appimage (Hungarian)
  - https://404.g-net.pl/2018/08/appimagelauncher/ (Polish)
  - http://www.edivaldobrito.com.br/integrador-appimagelauncher-no-linux/ (Portuguese)
  - http://linux-os.net/appimagelauncher-ejecuta-e-integra-facilmente-aplicaciones-en-appimage/ (Spanish)
    - same article also available here: https://blog.desdelinux.net/appimagelauncher-ejecuta-e-integra-facilmente-aplicaciones-en-appimage/
  - https://forum.pardus.org.tr/t/appimagelauncher-tek-tiklama-ile-appimage-sisteminizle-butunlestiriniz/11275 (Turkish)
  - https://linuxteamvietnam.us/cach-chay-ung-dung-dinh-dang-appimage-tren-linux/ (Vietnamese)

>     - contains a few wrong assumptions: as soon as AppImageLauncher is installed, you do not have to make AppImages executable, you can just double-click them (or otherwise open them, e.g., from your browsers' downloads)

Author: web-browser "Downloads", or clickable list of downloads?

Reviewer: Whatever UI the web browser offers to the user to directly open files they downloaded. IMO this is quite clear from the existing text.

Author: Quite is not what I'm going for. We need simple.

## Installation

### System-wide Installation

> AppImageLauncher integrates deeply into the systems. Therefore, an installation via native system packages is the preferred way to install AppImageLauncher. This way, AppImageLauncher's package can perform the necessary steps to have your system use it for all AppImage invocations.

Reviewer: The plural in "systems" makes no sense.

```suggestion
AppImageLauncher integrates deeply into the system. Therefore, an installation via native system packages is the preferred way to install AppImageLauncher. This way, AppImageLauncher's package can perform the necessary steps to have your system use it for all AppImage invocations.
```

> Compatibility table (likely incomplete, feel free to send PRs to add distributions)

Reviewer: Why would you remove "please"? Being polite here is not a mistake IMO.

Author: Pleonasm? It already says "feel free", which is an addition to "send PRs to __".

Reviewer: "Feel free" is more like "just" an offer. "Please" expresses that readers are really encouraged to create PRs. It's not a pleonasm of any kind.

Author: The please qualifies the feel free though. Sould be comma otherwise (?)

| Release filename | Build system | Compatible distributions (incomplete) |
| ---------------- | ------------ | ------------------------------------- |

> | `appimagelauncher-<version>.xenial_(amd64,i386).deb` | Ubuntu Xenial | Ubuntu Xenial (16.04), Debian Stretch (9), Netrunner 17.01 |

Reviewer: In my view, all the distros actually prefer to have these codenames be lower case. What's your reason to change the casing?

Author: Ubuntu doesn't, Debian does a bit of both. This doesn't reference the package repos.

  - https://404.g-net.pl/2018/08/appimagelauncher/ (Polish)
  - http://www.edivaldobrito.com.br/integrador-appimagelauncher-no-linux/ (Portuguese)
  - http://linux-os.net/appimagelauncher-ejecuta-e-integra-facilmente-aplicaciones-en-appimage/ (Spanish)
    - same article also available here: https://blog.desdelinux.net/appimagelauncher-ejecuta-e-integra-facilmente-aplicaciones-en-appimage/
  - https://forum.pardus.org.tr/t/appimagelauncher-tek-tiklama-ile-appimage-sisteminizle-butunlestiriniz/11275 (Turkish)
  - https://linuxteamvietnam.us/cach-chay-ung-dung-dinh-dang-appimage-tren-linux/ (Vietnamese)
    - contains a few wrong assumptions: as soon as AppImageLauncher is installed, you do not have to make AppImages executable, you can just double-click them (or otherwise open them, e.g., from your browsers' downloads)

## Installation

### System-wide Installation

AppImageLauncher integrates deeply into the system. Therefore, an installation via native system packages is the preferred way to install AppImageLauncher. This way, AppImageLauncher's package can perform the necessary steps to have your system use it for all AppImage invocations.

> Compatibility table (likely incomplete, feel free to send PRs to add distributions)

Author: ```suggestion
Compatibility table (likely incomplete, feel free to send PRs to add distributions).
```

Arch Linux, Manjaro, Antergos and Netrunner Rolling users can use AUR to install AppImageLauncher by installing [appimagelauncher-git](https://aur.archlinux.org/packages/appimagelauncher-git) (thanks @NuLogicSystems for setting up the build).

Other systems derived from the listed ones, such as for instance Linux Mint (Ubuntu), should support AppImageLauncher as well.
If they don't, please don't hesitate to create an issue.

> **Note:** If your system is not listed above as supported, feel free to request support in an issue to discuss adding support.

Reviewer: Same as above. Also, the now combined sentence contains the word "support" twice. Either restore the old form (and improve the wording this way), or shorten it even further:

```suggestion
**Note:** If your system is not listed above as supported, please feel free to open an issue to discuss adding support.
```

Arch Linux, Manjaro, Antergos and Netrunner Rolling users can use AUR to install AppImageLauncher by installing [appimagelauncher-git](https://aur.archlinux.org/packages/appimagelauncher-git) (thanks @NuLogicSystems for setting up the build).

Other systems derived from the listed ones, such as for instance Linux Mint (Ubuntu), should support AppImageLauncher as well.
If they don't, please don't hesitate to create an issue.

> **Note:** If your system is not listed above as supported, feel free to request support in an issue to discuss adding support.

Author: ```suggestion
**Note:** Open an issue if you want to have something added to the list of supported systems above.
```

Reviewer: Hmmm... This sentence kind of promises that any request will be implemented. That might be a bit "too much". "Something" is also a pretty dull wording.

```suggestion
**Note:** Please feel free to request support for additional distributions by [opening an issue](https://github.com/TheAssassin/AppImageLauncher/issues/new).
```

## How it works

AppImageLauncher is responsible for the desktop integration.

> When the user launches an AppImage, the software checks whether the AppImage has been integrated already.

Reviewer: These last two sections really need an update...

Author: I put them at the beginning before I saw the comment, so I agree wholeheartedly.

> When the user launches an AppImage, the software checks whether the AppImage has been integrated already.

Author: ```suggestion
When the user launches an AppImage, the software checks whether it has been integrated already.
```
